File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/GamesWiiU.java
package com.hiddenramblings.tagmo.amiibo.games;

import java.util.ArrayList;

public class GamesWiiU {

    public final GamesManager manager;
    public final long id;
    public final ArrayList<String> games;

    public GamesWiiU(GamesManager manager, long id, ArrayList<String> games) {
        this.manager = manager;
        this.id = id;
        this.games = games;
    }

    public String getStringList() {
        StringBuilder usage = new StringBuilder();
        for (String game : games) {
            if (usage.length() == 0)
                usage.append("  ");
            else
                usage.append(", ");
            usage.append(game);
        }
        return usage.toString();
    }

    public boolean hasUsage(String name) {
        return null != games && !games.isEmpty() && games.contains(name);
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/GamesWiiU.kt
package com.hiddenramblings.tagmo.amiibo.games

class GamesWiiU(val manager: GamesManager, val id: Long, val games: ArrayList<String?>?) {
    val stringList: String
        get() {
            val usage = StringBuilder()
            for (game in games!!) {
                if (usage.isEmpty()) usage.append("  ") else usage.append(", ")
                usage.append(game)
            }
            return usage.toString()
        }

    fun hasUsage(name: String?): Boolean {
        return games?.isNotEmpty() == true && games.contains(name)
    }
}